cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/PreferredLeaderElectionGoal.java

```diff
@@ -67,6 +95,11 @@ public boolean optimize(ClusterModel clusterModel, Set<Goal> optimizedGoals, Set
     return true;
   }
 
+  private boolean isPartitionUnderReplicated(Replica r) {
```

Comment:
Currently it seems a little difficult to understand which replica to pass to this function.

Would the change of the parameter from `Replica` to `Partition` and passing `r.topicPartition()` to this function make things more clear?